impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2009, Red Hat Middleware LLC, and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.shrinkwrap.impl.base.spec;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.NoSuchElementException;
+import java.util.logging.Logger;
+
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ConfigurationBuilder;
+import org.jboss.shrinkwrap.api.Domain;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Before;
+
+/**
+ * Base class for AddPackage* tests operating with a custom classloader.
+ *
+ * @author Falko Modler
+ * @author Bob McWhirter 
+ */
+public abstract class AddPackageTestBase {
+
+    /**
+     * Logger
+     */
+    private static final Logger LOG = Logger.getLogger(AddPackageTestBase.class.getName());
+
+    protected Domain domain;
+
+    private File tempFile;
```

Comment:
That's a very nice simplification, thanks for cleaning it up! I was wondering if we couldn't simply use [`TemporaryFolder` rule](http://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html) from JUnit to have this auto-cleanup done on our behalf?